src/poprox_storage/repositories/experiments.py

```diff
@@ -152,6 +154,31 @@ def fetch_experiment_phases(self, experiment_id: UUID) -> list[Phase]:
 
         return list(phases.values())
 
+    def fetch_experience_by_id(self, experience_id: str) -> Experience | None:
+        # Fetch an Experience by its ID from the experiences table.
+        experiences_table = self.tables.get("experiences")
+        if not experiences_table:
```

Comment:
You can assume the table will exist by this point in the code. If we failed to load it in the `__init__()` above, we will have already had an error by the time we get here